rust/src/x509/mod.rs

```diff
@@ -79,6 +94,37 @@ pub unsafe extern "C" fn rs_x509_get_subject(ptr: *const X509) -> *mut c_char {
     rust_string_to_c(subject)
 }
 
+#[no_mangle]
+pub unsafe extern "C" fn rs_x509_get_subjectaltname_len(ptr: *const X509) -> u16 {
+    if ptr.is_null() {
+        return 0;
+    }
+    let x509 = cast_pointer! {ptr, X509};
+    let san_list = x509.0.tbs_certificate.subject_alternative_name();
+    if let Ok(Some(sans)) = san_list {
+        // SAN length in a certificate is kept u16 following discussions at
+        // https://community.letsencrypt.org/t/why-sans-are-limited-to-100-domains-only
+        debug_validate_bug_on!(sans.value.general_names.len() == u16::MAX.into());
+        return sans.value.general_names.len() as u16;
```

Comment:
Hi, @chifflier !
I'm parsing the SubjectAlternativeNames with your `x509_parser`. In the parser, there seems to be [a vector of these](https://docs.rs/x509-parser/latest/src/x509_parser/extensions/mod.rs.html#478).
I was wondering if you could guide about how many SubjectAlternativeNames can be there in a TLS response? Is `u16` the correct type?
Thank you very much!